assets/js/homepage.js

```diff
@@ -62,22 +48,37 @@
 				var newr = Math.floor( ( 255 - r ) * brightness );
 				var newg = Math.floor( ( 255 - g ) * brightness );
 				var newb = Math.floor( ( 255 - b ) * brightness );
+				var color = 'rgb(' + newr + ', ' + newg + ', ' + newb + ')';
 
-				jQuery( homepageContent + ', .page-template-template-homepage .type-page.has-post-thumbnail h1' ).css( 'color', 'rgb(' + newr + ', ' + newg + ', ' + newb + ')' );
-				jQuery( '.page-template-template-homepage .type-page.has-post-thumbnail .entry-title, .page-template-template-homepage .type-page.has-post-thumbnail .entry-content' ).addClass( 'loaded' );
-
-				if ( brightness >= 30 ) {
-					jQuery( '.page-template-template-homepage .type-page.has-post-thumbnail .entry-title, .page-template-template-homepage .type-page.has-post-thumbnail .entry-content' ).css( 'text-shadow', '0 4px 30px rgba(0,0,0,.9)' );
-				}
+				homepageContent.style.color = color;
+				homepageContent.querySelectorAll('h1').forEach(function(h1) {
+					h1.style.color = color;
+				});
+				homepageContent.querySelectorAll('.entry-title, .entry-content').forEach(function(el) {
+					el.classList.add('loaded');
+					el.style.textShadow = brightness >= 30 ? '0 4px 30px rgba(0,0,0,.9)' : '';
+				});
 			}
 		});
 	});
 
 	/**
-	 * On window resize
-	 * Set hero content dimensions / layout
+	 * Set the hero component dimensions and positioning
 	 */
-	jQuery( window ).resize( function() {
-		homepageContentDimensions();
-	});
-} )();
+	function homepageContentDimensions() {
```

Comment:
We delay the update of the styles to the next frame, as this method might be called very frequently during window resize events.